scene/gui/tree.cpp

```diff
@@ -3340,7 +3340,8 @@ void Tree::gui_input(const Ref<InputEvent> &p_event) {
 		bool rtl = is_layout_rtl();
 
 		if (!mb->is_pressed()) {
-			if (mb->get_button_index() == MouseButton::LEFT) {
+			if (mb->get_button_index() == MouseButton::LEFT ||
+					mb->get_button_index() == MouseButton::RIGHT) {
```

Comment:
I wonder if this is necessary 🤔
This could probably support all buttons, since you have to check what button was used anyway.

Reply:
No what I mean is that maybe we could just allow any button, as there is no reason to prevent it I think.

Reply:
Ah, fair enough.

Reply:
@KoBeWi I wondered exactly the same thing.  I decided to leave in the restriction to be conservative.  No good scenarios why....   I thought maybe for the possibility of having some kind of consistent UI actions in the future that use a particular mouse button?

Reply:
But the signal emits either left or right button, so to use it you need to check what button you have received. Then it doesn't matter how many buttons it supports, as you already have filter out anything unnecessary.

Reply:
I agree.  I just wondered if Godot itself in the future might want to implement what happens with other buttons for consistency?   I didn't say it was a weak reason :)

I do think that this can of worms is a proposal, because it would affect ALL the "clicked" events everywhere.

Reply:
More support for your point:  we are already breaking compatibility, might as well make it nice and generic this time so we don't have to do that to people again after 4.0 is final.   So yeah, a proposal across all Control nodes to remove restrictions on which buttons are notified via signals?

Reply:
Let's just make a change here first. If there are other controls that report button index in signals which are restricted, we can evaluate it separately.

Reply:
Even this class has more than one signal like that.  Those were all changed in a previous PR to use button index (restricted) except this one was missed.  I fixed it because I need it to select shown columns in the Thread debugger UI :).   Are you suggesting to widen THIS PR to change more than that?

(sorted by me just now for clarification, not what the code has)
```
	ADD_SIGNAL(MethodInfo("column_title_clicked", PropertyInfo(Variant::INT, "column"), PropertyInfo(Variant::INT, "mouse_button_index")));
	ADD_SIGNAL(MethodInfo("item_mouse_selected", PropertyInfo(Variant::VECTOR2, "position"), PropertyInfo(Variant::INT, "mouse_button_index")));
	ADD_SIGNAL(MethodInfo("custom_item_clicked", PropertyInfo(Variant::INT, "mouse_button_index")));
	ADD_SIGNAL(MethodInfo("item_double_clicked"));
	ADD_SIGNAL(MethodInfo("button_clicked", PropertyInfo(Variant::OBJECT, "item", PROPERTY_HINT_RESOURCE_TYPE, "TreeItem"), PropertyInfo(Variant::INT, "column"), PropertyInfo(Variant::INT, "id"), PropertyInfo(Variant::INT, "mouse_button_index")));
	ADD_SIGNAL(MethodInfo("empty_clicked", PropertyInfo(Variant::VECTOR2, "position"), PropertyInfo(Variant::INT, "mouse_button_index")));

	ADD_SIGNAL(MethodInfo("item_selected"));
	ADD_SIGNAL(MethodInfo("cell_selected"));
	ADD_SIGNAL(MethodInfo("multi_selected", PropertyInfo(Variant::OBJECT, "item", PROPERTY_HINT_RESOURCE_TYPE, "TreeItem"), PropertyInfo(Variant::INT, "column"), PropertyInfo(Variant::BOOL, "selected")));
	ADD_SIGNAL(MethodInfo("item_edited"));
	ADD_SIGNAL(MethodInfo("item_custom_button_pressed"));
	ADD_SIGNAL(MethodInfo("item_collapsed", PropertyInfo(Variant::OBJECT, "item", PROPERTY_HINT_RESOURCE_TYPE, "TreeItem")));
	ADD_SIGNAL(MethodInfo("check_propagated_to_item", PropertyInfo(Variant::OBJECT, "item", PROPERTY_HINT_RESOURCE_TYPE, "TreeItem"), PropertyInfo(Variant::INT, "column")));
	ADD_SIGNAL(MethodInfo("custom_popup_edited", PropertyInfo(Variant::BOOL, "arrow_clicked")));
	ADD_SIGNAL(MethodInfo("item_activated"));
	ADD_SIGNAL(MethodInfo("nothing_selected"));
```

Reply:
> Are you suggesting to widen THIS PR to change more than that?

No. This is already an extra comment on the approval from KoBeWi, so this is by no means a requirement.

Reply:
Oh thank g*d*t. :)